test/Snowflake_Tests/polyglot-sources/snowflake-test-java-helpers/src/main/java/org/enso/snowflake_helpers/TestKeyGenerator.java

```diff
@@ -19,6 +19,13 @@
 import org.bouncycastle.operator.OutputEncryptor;
 
 public class TestKeyGenerator {
+
+  static {
+    if (Security.getProvider(BouncyCastleProvider.PROVIDER_NAME) == null) {
+      Security.addProvider(new BouncyCastleProvider());
+    }
+  }
```

Comment:
Ideally the `TestKeyGenerator` at some point would not be shipped with the production release as it is only needed for testing.

However I guess this bit may actually be worth keeping - as perhaps it initializes the cryptographic components that are then needed by Snowflake to be able to accept private keys.

If it is indeed needed to load private keys (would need to test by removing it and running the GUI trying to use some manually created key), then it could be worth to move this bit into `std-snowflake`.

Reply:
It is not being shipped with the release. It is only included when building NI for testing i.e., `ENSO_LAUNCHER=native,test`. Similarly with base tests' helper jar. I changed that in #12804.

Native Image will do this initialization at build time. This change is only added so that things work if you are running in JVM mode.

Reply:
Oh great to know.